### PR TITLE
Add interface to `lightning.qubit` MCM preprocessing

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Currently blocked by https://github.com/PennyLaneAI/pennylane/pull/5791. Device preprocessing is being moved inside the ML boundary, so the interface information needed for correct `hw-like` postselection is being lost when using `dynamic_one_shot`.

**Description of the Change:**
`qml.devices.preprocess.mid_circuit_measurements` accepts an additional `interface` kwarg, which is now being used in `LightningQubit.preprocess`.

**Benefits:**
Expected behaviour with `mcm_method="one-shot"`, `postselect_mode="hw-like"`, and `jax.jit`.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 pip~=22.0
-git+https://github.com/PennyLaneAI/pennylane.git@master
+git+https://github.com/PennyLaneAI/pennylane.git@inner-execute
```
Comment (mudit2812) [inline requirements-dev.txt:2]: Change this once https://github.com/PennyLaneAI/pennylane/pull/5791 is merged.

```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

**Comment by mudit2812** [comment]
[sc-65738]

**Review by multiphaseCFD — APPROVED** [review]
Looks good to me! Thanks @mudit2812 !

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -86,6 +86,9 @@
 * `lightning.qubit` and `lightning.kokkos` use `qml.ops.Conditional.base` instead of `qml.ops.Conditional.then_op`.
   [(#752)](https://github.com/PennyLaneAI/pennylane-lightning/pull/752)
 
+* `lightning.qubit`'s preprocessing for mid-circuit measurements correctly uses interface information.
```
Comment (maliasadi) [inline .github/CHANGELOG.md:89]:
```suggestion
* The preprocessing step in `lightning.qubit` now uses interface information to properly support the hardware-like postselection for mid-circuit measurements.
```

**Review by maliasadi — APPROVED** [review]
:raised_hands:

**Pull request merged**